docs/integration: recommend using stable paths in fstab and system.conf (#1271)
rauc/rauc: docs/integration.rst

Since changing the partition layout is hard or even impossible to change in the
field, make sure it meets both current and possible future requirements.

.. note::

  The ``/etc/fstab`` in your system's root FS (and RAUC's ``system.conf``)

>   should normally use stable paths to refer to the partitions or devices.

Reviewer: ```suggestion
  should normally use stable paths to refer to partitions or devices.
```

>   Especially a filesystem UUIDs (``UUID=<uuid>`` or

Reviewer: ```suggestion
  Especially filesystem UUIDs (``UUID=<uuid>`` or
```

  ``/dev/disk/by-uuid/<uuid>``) should *not* be used, as they are likely to be
  different after an update.

  Depending on your system design and firmware, good stable paths can be:

>   * plain device names: ``/dev/sda``, ``/dev/mmcblk0p1``, ``/dev/nvme0n1p1``, ``/dev/mapper/…``

Reviewer: Should we mention that these may depend on storage enumeration order in some cases?

  * topology-based symlinks: ``/dev/disk/by-path/…`` (preferable if available)

>   * partition-table-uuid-based symlinks: ``/dev/disk/by-partuuid/…`` (breaks if
>     using the same disk image on i.e. eMMC & SD)

Reviewer: ```suggestion
  * partition-table-UUID-based symlinks: ``/dev/disk/by-partuuid/…`` (breaks if
    using the same disk image on e.g. both eMMC & SD card)
```
